atlasdb-client/src/main/java/com/palantir/atlasdb/table/description/render/TableRendererV2.java

```diff
@@ -0,0 +1,422 @@
+        Preconditions.checkNotNull(packageName);
+        Preconditions.checkNotNull(namespace);
+        this.packageName = packageName;
+        this.namespace = namespace;
+    }
+
+    public String getClassName(String rawTableName, TableDefinition table) {
+        return Renderers.getClassTableName(rawTableName, table) + "SimpleTable";
+    }
+
+    public String render(String rawTableName, TableDefinition table) {
+        return new ClassRenderer(rawTableName, table).render();
+    }
+
+    private class ClassRenderer {
+        private final TableMetadata tableMetadata;
+        private final String rawTableName;
+        private final String tableName;
+
+        private final String tableClassName;
+        private final ClassName tableType;
+
+        private final String rowClassName;
+        private final ClassName rowType;
+
+        private final String rowResultClassName;
+        private final ClassName rowResultType;
+
+        private final String simpleTableName;
+        private final ClassName simpleTableType;
+
+        ClassRenderer(String rawTableName, TableDefinition table) {
+            Preconditions.checkArgument(
+                    Schemas.isTableNameValid(rawTableName),
+                    "Invalid table name " + rawTableName);
+            this.rawTableName = rawTableName;
+            this.tableName = Renderers.getClassTableName(rawTableName, table);
+            this.tableMetadata = table.toTableMetadata();
+
+            this.simpleTableName = this.tableName + "SimpleTable";
+            this.tableClassName = this.tableName + "Table";
+            this.rowClassName = this.tableName + "Row";
+            this.rowResultClassName = this.tableName + "RowResult";
+
+            this.tableType = ClassName.get(packageName, this.tableClassName);
+            this.simpleTableType = ClassName.get(packageName, this.simpleTableName);
+
+            this.rowType = this.tableType.nestedClass(this.rowClassName);
+            this.rowResultType = this.tableType.nestedClass(this.rowResultClassName);
+        }
+
+        public String render() {
+            JavaFile javaFile = JavaFile.builder(packageName, this.run())
+                    .indent("    ")
+                    .build();
+
+            return javaFile.toString();
+        }
+
+        private TypeSpec run() {
+            TypeSpec.Builder tableBuilder = TypeSpec.classBuilder(this.simpleTableName)
+                    .addAnnotation(AnnotationSpec.builder(Generated.class)
+                            .addMember("value", "$S", TableRendererV2.class.getName())
+                            .build())
+                    .addAnnotation(AnnotationSpec.builder(SuppressWarnings.class)
+                            .addMember("value", "$S", "all")
+                            .build())
+                    .addModifiers(Modifier.PUBLIC, Modifier.FINAL);
+
+            buildFactory(this::getFields, tableBuilder::addField);
```

Comment:
hmm, might be more straightforward to do `getFields().forEach(tableBuilder::addField)`